equational_theories/Completeness.lean

```diff
@@ -156,3 +156,71 @@ theorem Completeness' {α β} {Γ : Ctx α} {E : MagmaLaw β} (h : Γ ⊧ E) : N
 theorem Completeness {α} {Γ : Ctx α} {E : MagmaLaw α} (h : Γ ⊧ E) : Nonempty (Γ ⊢ E) :=
   match Completeness' h with
   | .intro x => .intro (derive_of_derive' x)
+
+#check ⟦3⟧
+
+#print Soundness'
+#print satisfiesPhi
+
+def FreeMagmaWithLaws.eval {α G} {Γ : Ctx α} (φ : α → G) [Magma G] (modelsG : G ⊧ Γ) :
+FreeMagmaWithLaws α Γ → G :=
+Quotient.lift (evalInMagma φ) (
+by
+  intros a b
+  simp [HasEquiv.Equiv, SetoidOfLaws, RelOfLaws]
```

Comment:
This non-terminal `simp` should be avoided. Please use the `simp only` output of `simp?`.